From e3e35f5f85ad4f173c87221f3837f9619bad85e5 Mon Sep 17 00:00:00 2001
From: Ferdinand Xu <cheng.a.xu@intel.com>
Date: Wed, 8 Jul 2015 20:58:22 -0400
Subject: [PATCH 0210/1164] CDH-29988: HIVE-11203: Beeline force option
 doesn't force execution when errors occurred in a
 script.(Ferdinand, reviewed by Xuefu Zhang)

---
 .../src/java/org/apache/hive/beeline/BeeLine.java  |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/BeeLine.java b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
index 92ae64c..d40dae0 100644
--- a/beeline/src/java/org/apache/hive/beeline/BeeLine.java
+++ b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
@@ -787,7 +787,7 @@ private int executeFile(String fileName) {
     FileInputStream initStream = null;
     try {
       initStream = new FileInputStream(fileName);
-      return execute(getConsoleReader(initStream), true);
+      return execute(getConsoleReader(initStream), !getOpts().getForce());
     } catch (Throwable t) {
       handleException(t);
       return ERRNO_OTHER;
-- 
1.7.9.5

